### PR TITLE
Ability to whitelist a key

### LINKED ISSUE #34: IP Whitelist?
Would it be any way to allow a specific set of IPs to not have any throttling at all? Haven't found any way to do that yet. Thanks!

### PR DESCRIPTION
**Pull request by TheAwiteb**
Closes https://github.com/AaronErhardt/actix-governor/issues/34 (No Breaking Changes)

### CODE REVIEW

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Looks good, but it'd be great if you could extend the tests to not only include the happy path.

**Comment by TheAwiteb** [comment]
> happy path

lol 😆. I will

**Comment by TheAwiteb** [comment]
In functional way 👨🏻‍💻

File: src/tests.rs [hunk 1]
```diff
@@ -627,40 +632,84 @@ async fn test_key_extraction_whitelisted_key_with_header() {
...
+            remaining: Option<&HeaderValue>,
+            after: Option<&HeaderValue>,
+            excepted_status: u16| {
+            assert_eq!(status, excepted_status);
+            assert_eq!(headers.get("x-ratelimit-whitelisted"), whitelisted);
+            assert_eq!(headers.get("x-ratelimit-limit"), limit);
+            assert_eq!(headers.get("x-ratelimit-remaining"), remaining);
+            assert_eq!(headers.get("x-ratelimit-after"), after);
+        }
+    };
```
Comment (AaronErhardt) [inline src/tests.rs]: Can't you just summarize this into once function? I don't understand why you need two closures here.

Also, FYI [happy path](https://en.wikipedia.org/wiki/Happy_path) is an actual term used in CS :)

Comment (TheAwiteb) [reply src/tests.rs]: > Also, FYI [happy path](https://en.wikipedia.org/wiki/Happy_path) is an actual term used in CS :)

Ohh 😆. New to me

**Review by AaronErhardt — APPROVED** [review]
Thanks!

**Comment by penso** [comment]
Awesome @TheAwiteb thanks!

**Pull request merged**